### DISCUSSION TITLE
Why are some table byte sizes highlighted?

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#16106] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/74785256/183453450-7e2ccb76-1330-40bc-b830-4bb5925c1e03.png)

I'm not understanding the meaning of the gray highlights around the M and the 1G... is this just an arbitrary sort of 'warning' about table data size?

I'm not even sure what to call that "column" (that displays the table size)... is it the "byte size" of the table?

Example of the 1G un-selected, and another one without the highlight being selected, to show the difference:

![image](https://user-images.githubusercontent.com/74785256/162036165-cd169fb0-53e0-4c72-b19e-2c69468369a9.png)

**Comment by e3dio** [comment, marked as answer]
Looks like it measures the size relative to the other tables, the biggest table being 100% width

**Reply by givihuda** [reply]
I'm not understanding this answer; OK, so it measures the size relative to other tables... I kind of assumed that much, but that doesn't explain why it's highlighting in gray what it is. Why is it highlighting the way it is in my screenshot?

**Reply by givihuda** [reply]
IE, why is it highlighting just the M in 270M but the whole 1G in 1G?

**Reply by ShadelessFox** [reply]
If you're asking why 1G is bigger than 270M is that because G (gigabyte) is 1024M (megabyte).

You can read more about units of measurement on the Wikipedia: https://en.m.wikipedia.org/wiki/Binary_prefix

**Reply by e3dio** [reply]
It's highlighting 100% of the `imb_cache` table and 27% of the `imb` table, it is 27% the size

**Reply by givihuda** [reply]
> If you're asking why 1G is bigger than 270M

No, I'm not asking why 1G is bigger than 270M... and I already understand how megabytes/gigabytes work, thank you.

I asked why the highlighting is the way it is. Why highlight in gray only the letter for Megabytes but both the number and the letter for Gigabyte(s)? What's the thinking behind this _very particular_ highlighting?

> It's highlighting 100% of the imb_cache table and 27% of the imb table, it is 27% the size

OK but I don't see how this answers my question about _how it decides what to highlight_, as in, why is it not consistently highlighting either both the number and letter or just the number or just the letter when the difference is an M or G number. I'm not seeing the _meaning_ of the differing highlighting.

**Reply by ShadelessFox** [reply]
It just paints a rectangle behind statistics (statistics is the object size, including digits and unit letters). Its width is the percentage relative to the biggest object in the folder.

Let's review this picture:
![](https://user-images.githubusercontent.com/74785256/162036165-cd169fb0-53e0-4c72-b19e-2c69468369a9.png)

In this case, the 48K (48 kilobytes) object happened to be so small in comparison to the 270MB (276480 kilobytes) object, that it literally occupies less than a single percent relative to it. Assuming that the maximum width of this rectangle is 50 pixels, multiplying it by such a small number and rounding it up will result in a one-pixel width rectangle, just like in your case.

**Reply by e3dio** [reply]
![image](https://user-images.githubusercontent.com/85405955/173384611-d42ab14a-f1da-401c-ad47-39d41a2bcde3.png)

**Reply by givihuda** [reply]
OK, interesting... I suppose having the bar start on the left would eliminate any ambiguity with it appearing to highlight a letter or a number.

**Reply by ShadelessFox** [reply]
It may look weird, too. Perhaps a small border around the text would resolve the issue...

**Reply by givihuda** [reply]
Having the bar start on the left is simpler and more intuitive and less graphically cluttered than a small border around the text.

**Reply by ShadelessFox** [reply]
I have done some tweaking and got the following result:

![image](https://user-images.githubusercontent.com/35821147/177660984-d4b63ccd-78c8-4049-af2d-797820594cd0.png)

I like how it looks (I changed colors for clarity). Without the border, it just floats somewhere and looks goofy.

Additionally, now it fills the rectangle based on the percent of the total size, not the biggest object in the containing folder.

WDYT?

**Reply by givihuda** [reply]
Sure, though I think people will prefer a lighter border (I think they won't like a border visually "cluttering" the numbers, but then I don't think many will even look at the numbers often).

Also, if the dark bar overlaps the number, they're almost the same color, so as to probably make the number hard to read.

If you do a border and a bar, then of course lighter and more contrasty colors would be better.

**Comment by LonwoLonwo** [comment]
Yes, @e3dio is totally right.

Grayed parts - are column statistics and they depend on table sizes.

You can disable this setting if you do not need it.

![2022-04-07 16_10_26-DBeaver Enterprise 21 3 3 - Playlist](https://user-images.githubusercontent.com/45152336/162206666-c294d0fa-38a0-4a25-afac-3bbec8219ccd.png)
